pkg/api/v20160330/validate.go

```diff
@@ -71,6 +71,15 @@ func (l *LinuxProfile) Validate() error {
 
 // Validate implements APIObject
 func (a *Properties) Validate() error {
+	if a.OrchestratorProfile == nil {
+		return fmt.Errorf("missing OrchestratorProfile")
+	}
```

Comment:
Currently all the defaults are set right before template generation which happens after validation. So not currently. It would make sense to have version specific defaults that we could call right before calling validate in apiloader.go.LoadContainerService.